crypto/aria/aria.c

```diff
@@ -7,15 +7,834 @@
  * https://www.openssl.org/source/license.html
  */
 
+/* 
+ * Copyright (C) 2017 National Security Research Institute. All Rights Reserved.
+ *
+ * Information for ARIA
+ *     http://210.104.33.10/ARIA/index-e.html (English)
+ *     http://seed.kisa.or.kr/ (Korean)
+ *
+ * Public domain version is distributed above.
+ */
+
 /* ====================================================================
  * Copyright (c) 2017 Oracle and/or its affiliates.  All rights reserved.
  */
 
-#include <assert.h>
 #include <openssl/e_os2.h>
-#include <string.h>
 #include "internal/aria.h"
 
+#include <assert.h>
+#include <string.h>
+#include <stddef.h>
+
+#ifndef OPENSSL_SMALL_FOOTPRINT
+
+/* Begin macro */
+
+/* rotation */
+#define rotl32(v, r) (((uint32_t)(v) << (r)) | ((uint32_t)(v) >> (32 - r)))
+#define rotr32(v, r) (((uint32_t)(v) >> (r)) | ((uint32_t)(v) << (32 - r)))
+
+#define bswap32(v)                                          \
+    (((v) << 24) ^ ((v) >> 24) ^                            \
+    (((v) & 0x0000ff00) << 8) ^ (((v) & 0x00ff0000) >> 8))
+
+#define GET_U8_BE(X, Y) ((uint8_t)((X) >> ((3 - Y) * 8)))
+#define GET_U32_BE(X, Y) ( \
+    ((uint32_t)((const uint8_t *)(X))[Y * 4    ] << 24) ^ \
+    ((uint32_t)((const uint8_t *)(X))[Y * 4 + 1] << 16) ^ \
+    ((uint32_t)((const uint8_t *)(X))[Y * 4 + 2] <<  8) ^ \
+    ((uint32_t)((const uint8_t *)(X))[Y * 4 + 3]      ))
+
+#define PUT_U32_BE(DEST, IDX, VAL) \
+    do { \
+        ((uint8_t *)(DEST))[IDX * 4    ] = GET_U8_BE(VAL, 0); \
+        ((uint8_t *)(DEST))[IDX * 4 + 1] = GET_U8_BE(VAL, 1); \
+        ((uint8_t *)(DEST))[IDX * 4 + 2] = GET_U8_BE(VAL, 2); \
+        ((uint8_t *)(DEST))[IDX * 4 + 3] = GET_U8_BE(VAL, 3); \
+    } while(0)
+
+#define COPY_U128_PTR(DEST, SRC) \
```

Comment:
You can replace this with memcpy. Contemporary compilers inline it, and some will even use SIMD load and store, so that whole operation can be performed in just two instructions.